### ISSUE TITLE
Appreciation and question: When will the code for DeCLIP-FViT / OV-DQUO be released

### ISSUE THREAD

**Issue by Justlovesmile** [xiaomoguhz/DeCLIP#10]
First of all, thank you for your excellent work. Your research is truly valuable and inspiring to the community. 
I have been experimenting with your methods by adapting the CLIP feature extraction approach from DeCLIP-CATSeg, as shown below:
```python
# code from DeCLIP_CATSeg/cat_seg/cat_seg_model.py
clip_features = self.sem_seg_head.predictor.clip_model.encode_dense(
    clip_images,
    normalize=True,
    keep_shape=False,
    mode='csa',
)
```
I applied a similar modification in the FViT codebase:
```python
# code from CLIPSelf-main/F-ViT/models/evaclip_vit.py
# x = visual.blocks[-1].forward_without_attn(x)      # [original code]
x = visual.blocks[-1].forward_without_rcffn(x, mode='csa')
```
Using the model weights from [DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0](https://huggingface.co/xiaomoguhzz/DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0), I trained the model for 3 epochs and achieved novel_AP50 of 38.2210 on the OV-COCO dataset. While the result is already impressive, it still falls short of the 41.1 novel_AP50 reported in your paper.

**May I ask if there are any other key modifications or details that I might be missing?**

Also, I would greatly appreciate it if you could share an estimated timeline for the open-sourcing of FViT and OV-DQUO. It would be extremely helpful for further research and reproduction efforts.

Thank you again, and I’m looking forward to your response.

**Comment by xiaomoguhz** [comment]
@Justlovesmile hi，Thank you for your attention to our work. Currently, I am preparing another paper, and once the work is completed, we will open-source the code for open-vocabulary object detection. Regarding performance issues, you may prioritize adjusting the `temp`, `beta`, and `alpha` parameters in F-ViT to better adapt to our backbone, as these parameters were originally designed to accommodate ClipSelf. This should be able to help you quickly. 
Best wishes

**Comment by xiaomoguhz** [comment]
In addition, **it is worth mentioning that this value is based on the experimental results of DeCLIP obtained through proposal distillation on F-ViT**, which can be specifically referenced in the original CLIPSelf paper. Therefore, you can also first obtain a DeCLIP through proposal distillation and then conduct experiments on F-ViT. You only need to change the `dataset_type` in DeCLIP to proposals_distill.

**Comment by Justlovesmile** [comment]
@xiaomoguhz Thank you so much for the quick and helpful response! I really appreciate your willingness to share details — it’s impressive how responsive and supportive you are. I’ll try out the suggestions you provided. Looking forward to the code release and your future work! 👍ヾ(≧▽≦*)o

**Comment by Justlovesmile** [comment]
May I also ask how to properly reproduce the DeCLIP-CATSeg results for open-vocabulary segmentation?

When I directly evaluate the pre-trained [CATSeg_DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0](https://huggingface.co/xiaomoguhzz/CATSeg_DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0), the results are very close to those reported in the paper — which is great! However, when I try to reproduce the results from scratch using the [DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0](https://huggingface.co/xiaomoguhzz/DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0) backbone weight, the performance on my side is quite poor.

Could you kindly let me know if there are any specific modifications required in the config or training setup (e.g., learning rate, number of epochs, loss weights, etc.) to successfully reproduce the results?

Here are the evaluation results I obtained from training logs:
```bash
##############################################################################
# Using CLIP_PRETRAINED: "EVA02-CLIP-B-16", CACHE_DIR: "DeCLIP_EVA-B_DINOv2-B_csa_0.05_2.0.pt"#
##############################################################################
[06/29 05:56:38] d2.engine.defaults INFO: Evaluation results for ade20k_150_test_sem_seg in csv format:
[06/29 05:56:38] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 05:56:38] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 05:56:38] d2.evaluation.testing INFO: copypaste: 1.8325,14.9668,3.0550,21.2411
[06/29 06:00:46] d2.engine.defaults INFO: Evaluation results for ade20k_full_sem_seg_freq_val_all in csv format:
[06/29 06:00:46] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:00:46] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:00:46] d2.evaluation.testing INFO: copypaste: 1.2392,14.4881,2.9301,18.9001
[06/29 06:02:09] d2.engine.defaults INFO: Evaluation results for voc_2012_test_sem_seg in csv format:
[06/29 06:02:09] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:02:09] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:02:09] d2.evaluation.testing INFO: copypaste: 1.0006,3.9614,5.0058,19.8982
[06/29 06:03:47] d2.engine.defaults INFO: Evaluation results for voc_2012_test_background_sem_seg in csv format:
[06/29 06:03:47] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:03:47] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:03:47] d2.evaluation.testing INFO: copypaste: 3.4641,46.1570,7.6369,53.0399
[06/29 06:08:08] d2.engine.defaults INFO: Evaluation results for context_59_test_sem_seg in csv format:
[06/29 06:08:08] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:08:08] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:08:08] d2.evaluation.testing INFO: copypaste: 2.5464,7.9555,5.1720,17.4202
[06/29 06:17:05] d2.engine.defaults INFO: Evaluation results for context_459_test_sem_seg in csv format:
[06/29 06:17:05] d2.evaluation.testing INFO: copypaste: Task: sem_seg
[06/29 06:17:05] d2.evaluation.testing INFO: copypaste: mIoU,fwIoU,mACC,pACC
[06/29 06:17:05] d2.evaluation.testing INFO: copypaste: 0.9356,7.0879,1.8343,14.5989
```

**Comment by Justlovesmile** [comment]
Apologies for the confusion — I just realized that the issue was on my side. I mistakenly downloaded the [stuff_maps](http://calvin.inf.ed.ac.uk/wp-content/uploads/data/cocostuffdataset/stuff_trainval2017.zip) version of the dataset, while CATSeg requires the [stuffthingmaps](http://calvin.inf.ed.ac.uk/wp-content/uploads/data/cocostuffdataset/stuffthingmaps_trainval2017.zip) version. /(ㄒoㄒ)/~~ Thank you again for your excellent work and sorry for the oversight!

**Comment by Justlovesmile** [comment]
> In addition, **it is worth mentioning that this value is based on the experimental results of DeCLIP obtained through proposal distillation on F-ViT**, which can be specifically referenced in the original CLIPSelf paper. Therefore, you can also first obtain a DeCLIP through proposal distillation and then conduct experiments on F-ViT. You only need to change the `dataset_type` in DeCLIP to proposals_distill.

@xiaomoguhz Hi, I am currently trying to reproduce the results on the LVIS dataset, and I would like to confirm some details about the hyperparameter settings you used. In particular, I am interested in the image size configurations, such as:

`--det-image-size` parameter in DeCLIP and `image_size` parameter in FViT

Could you kindly clarify the values you used for these (and possibly other key hyperparameters) when training or evaluating on LVIS? This information would be very helpful for reproducing the reported results.

Thank you very much for your time and assistance!
